### PR TITLE
Forbid negative scores in function_score query

### PR DESCRIPTION
**Pull request by mayya-sharipova**
- Throw an exception when scores are negative in field_value_factor
function
- Throw an exception when scores are negative in script_score
function

Relates to #33309

### CODE REVIEW

File: modules/lang-painless/src/test/resources/rest-api-spec/test/painless/30_search.yml [hunk 1]
```diff
@@ -442,3 +442,41 @@
...
+          catch: bad_request
+          search:
+              index: test
+              body:
+                  query:
+                      function_score:
+                          query:
+                              term:
+                                  test: value
+                          "functions": [{
```
Comment (jimczi) [inline modules/lang-painless/src/test/resources/rest-api-spec/test/painless/30_search.yml]: nit: indent

**Review by jimczi — APPROVED** [review]
LGTM. I wonder if we also need to check the `finalScore`
 computed by `FunctionScoreQuery#FunctionFactorScorer#scorer` ? It should not be possible to produce a negative score with the current checks but I could have missed something.

**Review by jimczi — COMMENTED** [review]
Can you also add a note in the [breaking changes](https://github.com/elastic/elasticsearch/blob/master/docs/reference/migration/migrate_7_0/search.asciidoc) ?

**Comment by rjernst** [comment]
This this only be made in 7.0, not 6.6? Maybe a deprecation warning in 6.6 instead?

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Pull request merged**